internal/helm-storage-backend/release_test.go

```diff
@@ -0,0 +1,497 @@
+					Namespace:     "other-ns",
+					ChartLocation: "http://example.com/charts",
+				}),
+			},
+			expErrMsg: "rpc error: code = NotFound desc = Helm release 'other-ns/test-release' for TypeInstance '123' was not found",
+		},
+		{
+			name: "should return internal error",
+			request: &pb.GetValueRequest{
+				TypeInstanceId: "123",
+				Context: mustMarshal(t, ReleaseContext{
+					Name:          releaseName,
+					Namespace:     "other-ns",
+					ChartLocation: "http://example.com/charts",
+				}),
+			},
+			internalError: errors.New("internal error"),
+			expErrMsg:     "rpc error: code = Internal desc = while creating Helm get release client: internal error",
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			// given
+			expHelmRelease := fixHelmRelease(releaseName, releaseNamespace)
+			expFlags := &genericclioptions.ConfigFlags{ClusterName: ptr.String("testing")}
+
+			mockConfigurationProducer := func(inputFlags *genericclioptions.ConfigFlags, inputDriver, inputNs string) (*action.Configuration, error) {
+				if test.internalError != nil {
+					return nil, test.internalError
+				}
+				producer := mockConfigurationProducer(t, expHelmRelease, expFlags, "secrets")
+				return producer(inputFlags, inputDriver, inputNs)
+			}
+
+			svc, err := NewReleaseHandler(logger.Noop(), expFlags)
+			svc.actionConfigurationProducer = mockConfigurationProducer
+			require.NoError(t, err)
+
+			// when
+			outVal, gotErr := svc.GetValue(context.Background(), test.request)
+
+			// then
+			assert.EqualError(t, gotErr, test.expErrMsg)
+			assert.Nil(t, outVal)
+		})
+	}
+}
+
+func TestRelease_OnCreate_Success(t *testing.T) {
+	// given
+	const (
+		releaseName      = "test-create-release"
+		releaseNamespace = "test-create-namespace"
+		releaseDriver    = "configmap"
+		chartLocation    = "http://example.com/charts"
+	)
+	expHelmRelease := fixHelmRelease(releaseName, releaseNamespace)
+	expFlags := &genericclioptions.ConfigFlags{ClusterName: ptr.String("testing")}
+	mockConfigurationProducer := mockConfigurationProducer(t, expHelmRelease, expFlags, releaseDriver)
+
+	givenReq := &pb.OnCreateRequest{
+		TypeInstanceId: "42",
+		Context: mustMarshal(t, ReleaseContext{
+			Name:          releaseName,
+			Namespace:     releaseNamespace,
+			ChartLocation: chartLocation,
+			Driver:        ptr.String(releaseDriver),
+		}),
+	}
+
+	svc, err := NewReleaseHandler(logger.Noop(), expFlags)
+	svc.actionConfigurationProducer = mockConfigurationProducer
+	require.NoError(t, err)
+
+	// when
+	gotOut, gotErr := svc.OnCreate(context.Background(), givenReq)
+
+	// then
+	assert.NoError(t, gotErr)
+	assert.Empty(t, gotOut)
+}
+
+func TestRelease_OnCreate_Failures(t *testing.T) {
+	// globally given
+	const (
+		releaseName      = "test-release"
+		releaseNamespace = "test-namespace"
+	)
+	tests := []struct {
+		name string
+
+		request       *pb.OnCreateRequest
+		internalError error
```

Comment:
This is not used anywhere in any of the test cases

Reply:
fixed